dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/imports/HarvestedJsonParser.java

```diff
@@ -330,100 +280,133 @@ private Date parseTime(String str) throws ParseException {
         return Collections.singletonList(cvv);
     }
 
-    private List<DatasetField> parseCompoundValue(DatasetFieldType compoundType, JsonObject json, Boolean testType) throws JsonParseException {
-        List<ControlledVocabularyException> vocabExceptions = new ArrayList<>();
+    private List<DatasetField> parseCompoundValue(DatasetFieldType compoundType, JsonObject json) {
         List<DatasetField> vals = new LinkedList<>();
-        if (compoundType.isAllowMultiples()) {
-            try {
-                json.getJsonArray("value").getValuesAs(JsonObject.class);
-            } catch (ClassCastException cce) {
-                throw new JsonParseException("Invalid values submitted for " + compoundType.getName() + ". It should be an array of values.");
-            }
 
-            JsonArray value = json.getJsonArray("value");
-            System.out.println(value);
+        try {
+            if (compoundType.isAllowMultiples()) {
+                try {
+                    vals.addAll(parseCompoundMultiValueField(compoundType, json));
+                } catch (ClassCastException | NullPointerException exception) {
+                    vals.add(parseCompoundSingleValueField(compoundType, json));
```

Comment:
IMO instead of consecutive catching of `ClassCastException` and `NullPointerException` we should remove the cause why this situation happens and act accordingly. When we catch everywhere this kind of exceptions we kind of losing control on what are considering "normal" situation and what is not "normal" situation. We are basically telling: well if this method fails then maybe the other will not - it feels more like guessing than an actual coding. Catching NPE is especially dangerous because it can easily hide some defect's in the code that will be hard to track

Reply:
TBH I don't really know how to accomplish that.. 
Can you enlighten me how should I properly check if a given value is an array or a string?

Reply:
This is something that we already had in original code in one place:
`if (dft.isAllowMultiples() && json.get("value").getValueType().equals(JsonValue.ValueType.ARRAY))`
So: `JsonValue` have a method `.getValueType()`